ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/ForeignKeyTableTableJoinBuilderTest.java

```diff
@@ -0,0 +1,315 @@
+  }
+
+  @Test
+  public void shouldDoInnerJoinOnNonKey() {
+    // Given:
+    givenInnerJoin(left, JOIN_COLUMN);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTable).join(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(1)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldDoInnerJoinOnKey() {
+    // Given:
+    givenInnerJoin(left, L_KEY);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTable).join(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(2)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldDoInnerJoinOnSubKey() {
+    // Given:
+    givenInnerJoin(leftMultiKey, L_KEY_2);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTableMultiKey).join(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(3)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA_MULTI_KEY.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldReturnCorrectSchema() {
+    // Given:
+    givenInnerJoin(left, JOIN_COLUMN);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(
+        result.getSchema(),
+        is(LogicalSchema.builder()
+            .keyColumns(LEFT_SCHEMA.key())
+            .valueColumns(LEFT_SCHEMA.value())
+            .valueColumns(RIGHT_SCHEMA.value())
+            .build()
+        )
+    );
+  }
+
+  @Test
+  public void shouldReturnCorrectSchemaMultiKey() {
+    // Given:
+    givenInnerJoin(leftMultiKey, L_KEY);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(
+        result.getSchema(),
+        is(LogicalSchema.builder()
+            .keyColumns(LEFT_SCHEMA_MULTI_KEY.key())
+            .valueColumns(LEFT_SCHEMA_MULTI_KEY.value())
+            .valueColumns(RIGHT_SCHEMA.value())
+            .build())
+    );
+  }
+
+  // not sure what "legacy schema" is and if we need to support it?
```

Comment:
I don't think this is necessary. `LEGACY_KEY_COL` refers to the old default output key column name for joins. For FK joins, the output key columns always match the left source key columns.